### PR TITLE
remove solc_args

### PR DESCRIPTION
**Pull request by rahav-certora**
Naming convention:
 - PRs for features that are in design should have the "proposal" label
 - PRs for features that haven't landed yet should have the "future" label
 - PRs for upcoming releases should have the "release" label
 - PRs with new documentation for existing features should have the "existing feature" label

Before requesting review:
 - Make sure there is a ticket in the DOC board in Jira
 - Make sure CI is passing
   - Spell check failure may require adding backticks around code or updating `spelling_wordlist.txt`
   - See `README.md` for information about style and markdown syntax
   - If the CI Details link gives a 404, you need to log in to readthedocs.com
 - Add link to generated documentation here
   - you can find this by following the read the docs link from the CI check
 - Ask for help in #documentation

Jira ticket: TODO
Link to generated documentation: TODO

### CODE REVIEW

File: docs/prover/cli/options.md [hunk 1]
```diff
@@ -251,19 +251,6 @@ Whenever you want to use a Solidity compiler executable with a non-default name.
 **Example**  
 `certoraRun Bank.sol --verify Bank:Bank.spec --solc solc8.1`
 
-### `--solc_args`
```
Comment (mdgeorge4153) [inline docs/prover/cli/options.md:254]: I don't understand, was this option removed?  It is still reported in `certoraRun --help`

**Review by mdgeorge4153 — CHANGES_REQUESTED** [review]

File: docs/prover/cli/options.md [hunk 2]
```diff
@@ -251,19 +251,6 @@ Whenever you want to use a Solidity compiler executable with a non-default name.
...
-
-**What does it do?**
```
Comment (mdgeorge4153) [inline docs/prover/cli/options.md:256]: I might leave this section in but replace it with
````
```{versionchanged} 4.0
Support for `--solc_args` has been removed; see {ref}`--optimize`, ...
```
````

File: docs/prover/cli/options.md (hunk 1 above)
Comment (rahav-certora) [reply docs/prover/cli/options.md:254]: This option was removed. Once merged this option will not be listed in --help

File: docs/prover/cli/options.md (hunk 2 above)
Comment (rahav-certora) [reply docs/prover/cli/options.md:256]: I am not sure it is a good idea. From what I see `--solc_args` was not used for long time by customers. It is only for solc options that are not supported explicitly  like --solc_allow_path, solc_optimize, solc_via_ir etc,  and it seems that what we provide gives and gave good coverage

**Review by mdgeorge4153 — APPROVED** [review]

**Pull request merged**